code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -255,6 +256,10 @@ GLOBAL_LIST_EMPTY(roundstart_station_closets)
 	. += emissive_appearance(icon, "locked", src, alpha = src.alpha)
 	. += locked ? "locked" : "unlocked"
 
+/obj/structure/closet/update_atom_colour()
+	. = ..()
+	update_appearance()
+
```

Comment:
why? if we're gonna do this shouldn't we do this for everything?

Reply:
forgor